dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -119,40 +123,32 @@ public static Builder builder() {
 
     @NotThreadSafe
     public static final class Builder {
-        private static final ImmutableListMultimap<String, String> EMPTY =
-                new ImmutableListMultimap.Builder<String, String>().build();
 
-        @Nullable
-        private ListMultimap<String, String> headerParams;
+        @SuppressWarnings("UnnecessaryLambda") // Avoid unnecessary allocation
+        private static final com.google.common.base.Supplier<List<String>> MAP_VALUE_FACTORY = () -> new ArrayList<>(1);
 
-        @Nullable
-        private ImmutableListMultimap.Builder<String, String> queryParams;
+        private static final int MUTABLE_HEADERS = 1;
```

Comment:
MUTABLE_HEADERS_BITMASK ?

Reply:
Ya, these names are all awful, sorry. Will update them in a moment!